cli/src/lib.rs

```diff
@@ -3707,8 +3709,7 @@ fn create_idl_account(
 ) -> Result<Pubkey> {
     // Misc.
     let idl_address = IdlAccount::address(program_id);
-    let keypair = solana_sdk::signature::read_keypair_file(keypair_path)
-        .map_err(|_| anyhow!("Unable to read keypair file"))?;
+    let keypair = get_keypair(cfg.provider.wallet.to_string())?;
```

Comment:
This should use `keypair_path` instead:

```suggestion
    let keypair = get_keypair(keypair_path)?;
```

Reply:
This doesn't seem resolved. I'm not suggesting to provide `WalletPath` type here, `keypair_path` is `&str`. If you change the `path` parameter type of `get_keypair` function to `&str`, it should work.

Reply:
Yep, I wanted to chat to you about how to proceed.  Got it re: `&str`, done! Added a changelog too.